deployment/helm/templates/custom-configuration.yaml

```diff
@@ -7,26 +7,61 @@ spec:
   discoveryHandler: 
     name: {{ required "A custom.configuration.discoveryHandlerName is required." .Values.custom.configuration.discoveryHandlerName }}
     discoveryDetails: {{ .Values.custom.configuration.discoveryDetails }}
-  {{- if .Values.custom.configuration.brokerPod.image.repository }}
-  brokerPodSpec:
-    containers:
-    - name: {{ .Values.custom.configuration.name }}-broker
-      image: {{ printf "%s:%s" .Values.custom.configuration.brokerPod.image.repository .Values.custom.configuration.brokerPod.image.tag | quote }}
-      {{- with .Values.custom.configuration.pullPolicy }}
-      imagePullPolicy: {{ . }}
+  {{- if or .Values.custom.configuration.brokerPod.image.repository .Values.custom.configuration.brokerJob.image.repository }}
+  {{- /* Only add brokerSpec if a broker image is provided */}}
+  brokerSpec:
```

Comment:
so current template doesn't prevent a user from specifying both a podspec and a jobspec, which I believe is not supported. correct?
We may want to add that to the validating webhook.

Reply:
I believe it does prevent it. The configuration definition should disallow the definition of both brokerPodSpec and brokerJobSpec as the brokerSpec property [is an enum](https://github.com/kate-goldenring/akri/blob/strategy-jobs/shared/src/akri/configuration.rs#L39).

Reply:
oh i see what you're saying. The helm templates could construct a config with both a PodSpec and JobSpec. It would fail to be applied to the cluster because it wouldnt match the CRD definition, though

Reply:
added tests https://github.com/project-akri/akri/pull/437/commits/198ca4f365232f9006b364c4f49d0716587345d7